### ISSUE TITLE
Task blocks are considered a violation of `only-builtins` since 6.1.0

### ISSUE THREAD

**Issue by Victoremepunto** [ansible/ansible-lint#2186]
not sure if this could be related to  #1894  

Tested with ansible-lint versions since `6.0.2` up to current latest stable release (`6.2.2`) and current latest `master` (951a8284b77c2602a18de73fef1ef2c9e36cc497). The bug apparently appeared on `6.1.0` and it's not present in `6.0.2`

##### Summary

When running `only-builtins` against a specific playbook that contains rules within a block, the block is flagged as a `only-builtins` rule offender. The tasks within the block are correctly evaluated against `only-builtins`, but the block is always marked as violating the rule.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
####################### first environment where the bug appeared:
ansible --version
ansible [core 2.12.4.post0]
  config file = None
  configured module search path = ['/home/runner/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.8/site-packages/ansible
  ansible collection location = /home/runner/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.8.12 (default, Sep 21 2021, 00:10:52) [GCC 8.5.0 20210514 (Red Hat 8.5.0-3)]
  jinja version = 3.1.1
  libyaml = True


ansible-lint --version
WARNING: PATH altered to include /usr/bin
ansible-lint 6.1.0 using ansible 2.12.4.post0

################# last environment working version:
ansible --version
ansible [core 2.12.4.post0]
  config file = None
  configured module search path = ['/home/runner/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.8/site-packages/ansible
  ansible collection location = /home/runner/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.8.12 (default, Sep 21 2021, 00:10:52) [GCC 8.5.0 20210514 (Red Hat 8.5.0-3)]
  jinja version = 3.1.1
  libyaml = True

ansible-lint --version
WARNING: PATH altered to include /usr/bin
ansible-lint 6.0.2 using ansible 2.12.4.post0
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

Fedora 36, running on a [creator-ee](https://github.com/ansible/creator-ee/) with rootless podman (ver. 4.1.0)

##### STEPS TO REPRODUCE

- prepare a playbook like:

```yaml
# foobar.yaml
- name: Foo playbook
  hosts: web
  become: true
  tags:
    - some_tag
  tasks:
    - name: A block
      block:
        - name: a task that PASSES only-builtin
          lineinfile:
            key: value
        - name: a task that FAILS only-builtin rule check
          firewalld:
            key: value
```

- prepare a config file (only-builtin is experimental so requires opt-in):

```yaml
#ansible-lint-config.yml

enable_list:
  - only-builtins
```

Create a container for running the test:

```console (paste below)
podman run --rm -d --name ansible-lint -v $PWD/ansible-lint-config.yml:/config/ansible-lint-config.yml -v "$PWD/foobar.yaml:/content/foobar.yaml'  quay.io/ansible/creator-ee:v0.5.2 sleep infinity
```

Run the commands into the container

```
# podman exec -it ansible-lint bash

bash-4.4# ansible-lint /content/foobar.yaml -t only-builtins  -c /config/ansible-lint-config.yml -vvvv
```

##### Desired Behavior

`only-builtins` should've only failed to evaluate "a task that FAILS only-builtin rule check" 

##### Actual Behavior

Two `only-builtins` warnings are raised, the expected offending task, and the task block.

```
bash-4.4# ansible-lint /content/foobar.yaml -t only-builtins  -c /config/ansible-lint-config.yml -vvvv
DEBUG    Logging initialized to level 10
DEBUG    Options: Namespace(cache_dir='/home/runner/.cache/ansible-compat/b79606', colored=True, config_file='/config/ansible-lint-config.yml', configured=True, cwd=PosixPath('/runner'), display_relative_path=True, enable_list=['only-builtins'], exclude_paths=['.cache', '.git', '.hg', '.svn', '.tox'], extra_vars=None, format='rich', kinds=[{'jinja2': '**/*.j2'}, {'jinja2': '**/*.j2.*'}, {'yaml': '.github/**/*.{yaml,yml}'}, {'text': '**/templates/**/*.*'}, {'execution-environment': '**/execution-environment.yml'}, {'ansible-lint-config': '**/.ansible-lint'}, {'ansible-lint-config': '**/.config/ansible-lint.yml'}, {'ansible-navigator-config': '**/ansible-navigator.{yaml,yml}'}, {'inventory': '**/inventory/**.yml'}, {'requirements': '**/meta/requirements.yml'}, {'galaxy': '**/galaxy.yml'}, {'reno': '**/releasenotes/*/*.{yaml,yml}'}, {'playbook': '**/playbooks/*.{yml,yaml}'}, {'playbook': '**/*playbook*.{yml,yaml}'}, {'role': '**/roles/*/'}, {'tasks': '**/tasks/**/*.{yaml,yml}'}, {'handlers': '**/handlers/*.{yaml,yml}'}, {'vars': '**/{host_vars,group_vars,vars,defaults}/**/*.{yaml,yml}'}, {'test-meta': '**/tests/integration/targets/*/meta/main.{yaml,yml}'}, {'meta': '**/meta/main.{yaml,yml}'}, {'meta-runtime': '**/meta/runtime.{yaml,yml}'}, {'arg_specs': '**/roles/**/meta/argument_specs.{yaml,yml}'}, {'yaml': '.config/molecule/config.{yaml,yml}'}, {'requirements': '**/molecule/*/{collections,requirements}.{yaml,yml}'}, {'yaml': '**/molecule/*/{base,molecule}.{yaml,yml}'}, {'requirements': '**/requirements.yml'}, {'playbook': '**/molecule/*/*.{yaml,yml}'}, {'yaml': '**/{.ansible-lint,.yamllint}'}, {'yaml': '**/*.{yaml,yml}'}, {'yaml': '**/.*.{yaml,yml}'}], lintables=['/content/foobar.yaml'], listrules=False, listtags=False, loop_var_prefix=None, mock_modules=[], mock_roles=[], offline=False, parseable=False, progressive=False, project_dir='/config', quiet=False, rules={}, rulesdir=[], rulesdirs=['/usr/local/lib/python3.8/site-packages/ansiblelint/rules'], skip_action_validation=True, skip_list=[], tags=['only-builtins'], use_default_rules=False, var_naming_pattern=None, verbosity=4, version=False, warn_list=['experimental', 'role-name'], write_list=[])
DEBUG    /runner
INFO     Set ANSIBLE_LIBRARY=/home/runner/.cache/ansible-compat/527aa9/modules:/home/runner/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/home/runner/.cache/ansible-compat/527aa9/collections:/home/runner/.ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/home/runner/.cache/ansible-compat/527aa9/roles:/home/runner/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles
DEBUG    Loading rules from /usr/local/lib/python3.8/site-packages/ansiblelint/rules
INFO     Set ANSIBLE_LIBRARY=/home/runner/.cache/ansible-compat/527aa9/modules:/home/runner/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/home/runner/.cache/ansible-compat/527aa9/collections:/home/runner/.ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/home/runner/.cache/ansible-compat/527aa9/roles:/home/runner/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles
DEBUG    Effective yamllint rules used: {'braces': {'level': 'error', 'forbid': False, 'min-spaces-inside': 0, 'max-spaces-inside': 1, 'min-spaces-inside-empty': -1, 'max-spaces-inside-empty': -1}, 'brackets': {'level': 'error', 'forbid': False, 'min-spaces-inside': 0, 'max-spaces-inside': 0, 'min-spaces-inside-empty': -1, 'max-spaces-inside-empty': -1}, 'colons': {'level': 'error', 'max-spaces-before': 0, 'max-spaces-after': 1}, 'commas': {'level': 'error', 'max-spaces-before': 0, 'min-spaces-after': 1, 'max-spaces-after': 1}, 'comments': {'level': 'warning', 'require-starting-space': True, 'ignore-shebangs': True, 'min-spaces-from-content': 1}, 'comments-indentation': False, 'document-end': False, 'document-start': False, 'empty-lines': {'level': 'error', 'max': 2, 'max-start': 0, 'max-end': 0}, 'empty-values': False, 'hyphens': {'level': 'error', 'max-spaces-after': 1}, 'indentation': {'level': 'error', 'spaces': 'consistent', 'indent-sequences': True, 'check-multi-line-strings': False}, 'key-duplicates': {'level': 'error'}, 'key-ordering': False, 'line-length': {'level': 'error', 'max': 160, 'allow-non-breakable-words': True, 'allow-non-breakable-inline-mappings': False}, 'new-line-at-end-of-file': {'level': 'error'}, 'new-lines': {'level': 'error', 'type': 'unix'}, 'octal-values': False, 'quoted-strings': False, 'trailing-spaces': {'level': 'error'}, 'truthy': {'level': 'warning', 'allowed-values': ['true', 'false'], 'check-keys': True}}
WARNING  Overriding detected file kind 'yaml' with 'playbook' for given positional argument: /content/foobar.yaml
INFO     Executing syntax check on /content/foobar.yaml (0.52s)
DEBUG    Examining /content/foobar.yaml of type playbook
WARNING  Listing 2 violation(s) that are fatal
only-builtins: Use only builtin actions.
../content/foobar.yaml:7 Task/Handler: A block

only-builtins: Use only builtin actions.
../content/foobar.yaml:12 Task/Handler: a task that FAILS only-builtin rule check

You can skip specific rules or tags by adding them to your configuration file:
# .config/ansible-lint.yml
warn_list:  # or 'skip_list' to silence them completely
  - experimental  # all rules tagged as experimental

Finished with 0 failure(s), 2 warning(s) on 1 files.
```

**Comment by ssbarnea** [comment]
PR welcomed ;)

**Comment by evgeni** [comment]
No time for a PR (or tests… you pick ;)) but the following should fix it:

```patch
diff --git src/ansiblelint/rules/only_builtins.py src/ansiblelint/rules/only_builtins.py
index 74c33ec..3402ed3 100644
--- src/ansiblelint/rules/only_builtins.py
+++ src/ansiblelint/rules/only_builtins.py
@@ -5,6 +5,7 @@ from typing import Any, Dict, Optional, Union
 from ansiblelint.file_utils import Lintable
 from ansiblelint.rules import AnsibleLintRule
 
+from ansiblelint.skip_utils import is_nested_task
 # fqcn_builtins was added in 5.1.0 as FQCNBuiltinsRule, renamed to fqcn_builtins in 6.0.0
 from ansiblelint.rules.fqcn_builtins import builtins
 
@@ -24,7 +25,7 @@ class OnlyBuiltinsRule(AnsibleLintRule):
             "ansible.builtin."
         )
         non_fqcn_builtin = task["action"]["__ansible_module_original__"] in builtins
-        return not fqcn_builtin and not non_fqcn_builtin
+        return not fqcn_builtin and not non_fqcn_builtin and not is_nested_task(task)
 
 
 # testing code to be loaded only with pytest or when executed the rule file
```

**Comment by evgeni** [comment]
> not sure if this could be related to #1894

I don't think it is.

> Tested with ansible-lint versions since `6.0.2` up to current latest stable release (`6.2.2`) and current latest `master` ([951a828](https://github.com/ansible/ansible-lint/commit/951a8284b77c2602a18de73fef1ef2c9e36cc497)). The bug apparently appeared on `6.1.0` and it's not present in `6.0.2`

6.1.0 was the version that introduced the `only-builtins` rule, so it makes sense that no prior version is affected, and all later are.

PR is open and waiting on review.

**Comment by Victoremepunto** [comment]
> > not sure if this could be related to #1894
> 
> I don't think it is.

Thank you for confirming ! Yeah I looked through the open issues and that was the only one that seemed it could have something to do. 

> > Tested with ansible-lint versions since `6.0.2` up to current latest stable release (`6.2.2`) and current latest `master` ([951a828](https://github.com/ansible/ansible-lint/commit/951a8284b77c2602a18de73fef1ef2c9e36cc497)). The bug apparently appeared on `6.1.0` and it's not present in `6.0.2`
> 
> 6.1.0 was the version that introduced the `only-builtins` rule, so it makes sense that no prior version is affected, and all later are.

Well, I tested `only-builtins` with `6.0.2` , using it as a custom rule: I grabbed the `only-builtins` rule file from `master` and used it as a `custom` rule in, back then, the latest stable version, and it actually looked like it was working, it detected the tasks using non-core modules flawlessly, even within task blocks. 

It was very weird to see that once it was released it was no longer behaving as expected. 

> PR is open and waiting on review.

Thank you !! I'll try to review it myself and see how it works.
